packages/vite/src/node/config.ts

```diff
@@ -663,7 +662,44 @@ export async function resolveConfig(
       ...config.experimental
     }
   }
-  const resolved: ResolvedConfig = Object.assign(config, resolvedConfig)
+  const resolved: ResolvedConfig = {
+    ...config,
+    ...resolvedConfig
+  }
+
+  ;(resolved.plugins as Plugin[]) = await resolvePlugins(
+    resolved,
+    prePlugins,
+    normalPlugins,
+    postPlugins
+  )
+
+  const workerResolved: ResolvedConfig = {
+    ...workerConfig,
+    ...resolvedConfig,
+    isWorker: true,
```

Comment:
```suggestion
  const workerResolved: ResolvedConfig = {
    ...config,
    ...workerConfig,
    ...resolvedConfig,
    isWorker: true,
```

Following my comment above, if we dont want user plugins to affect the worker config, maybe we would do something like this 🤔